src/dos/dos_keyboard_layout.cpp

```diff
@@ -732,35 +837,16 @@ Bitu keyboard_layout::read_codepage_file(const char* codepage_file_name, Bit32s
 
 	if (!strcmp(cp_filename,"auto")) {
 		// select matching .cpi-file for specified codepage
-		switch (codepage_id) {
-			case 437:	case 850:	case 852:	case 853:	case 857:	case 858:	
-						sprintf(cp_filename, "EGA.CPI"); break;
-			case 775:	case 859:	case 1116:	case 1117:
-						sprintf(cp_filename, "EGA2.CPI"); break;
-			case 771:	case 772:	case 808:	case 855:	case 866:	case 872:
-						sprintf(cp_filename, "EGA3.CPI"); break;
-			case 848:	case 849:	case 1125:	case 1131:	case 61282:
-						sprintf(cp_filename, "EGA4.CPI"); break;
-			case 737:	case 851:	case 869:
-						sprintf(cp_filename, "EGA5.CPI"); break;
-			case 113:	case 899:	case 59829:	case 60853:
-						sprintf(cp_filename, "EGA6.CPI"); break;
-			case 58152:	case 58210:	case 59234:	case 60258:	case 62306:
-						sprintf(cp_filename, "EGA7.CPI"); break;
-			case 770:	case 773:	case 774:	case 777:	case 778:
-						sprintf(cp_filename, "EGA8.CPI"); break;
-			case 860:	case 861:	case 863:	case 865:
-						sprintf(cp_filename, "EGA9.CPI"); break;
-			case 667:	case 668:	case 790:	case 867:	case 991:	case 57781:
-						sprintf(cp_filename, "EGA10.CPI"); break;
-			default:
-				LOG_MSG("No matching cpi file for codepage %i",codepage_id);
-				return KEYB_INVALIDCPFILE;
+		auto i = get_CPX_file_id(codepage_id);
```

Comment:
`const auto i` ... 
(If it's read-only; I'm just looking at diffs though, so can't tell if we modify `i` later on)